src/lib/transition_frontier/persistent_frontier/diff_buffer.ml

```diff
@@ -95,12 +93,9 @@ let create ~(constraint_constants : Genesis_constants.Constraint_constants.t)
   t.timer <- Some timer ;
   t
 
-let write t ~diffs ~hash_transition =
-  let open Frontier_hash in
+let write t ~diffs =
   if t.closed then failwith "attempt to write to diff buffer after closed" ;
-  if not (Frontier_hash.equal t.target_hash hash_transition.source) then
-    failwith "invalid hash transition received by persistence buffer" ;
-  t.target_hash <- hash_transition.target ;
+  let (`Unprocessed diffs) = Worker.make_immediate_progress t.worker diffs in
```

Comment:
It might make more sense to make the diff buffer itself aware of the 2 different queues (regular diffs vs root transition diffs) rather than pushing that logic down into the worker itself. Then, the code here would read like:

```ocaml
let regular_diffs, root_transition_diffs = sort_diffs diffs in
List.iter diffs ~f:(DynArray.add t.regular_diffs) ;
Queue.enqueue_all t.root_transition_diffs root_transition_diffs ;
Worker.make_immediate_progress t.worker t.root_transition_diffs ;
(* ... *)
```

This would help to cleanup some of the worker code a little as well.

Reply:
Yeah, I agree it's confusing for that code not to be in the diff buffer. My thinking in doing it the way I did was maintaining the possibility for the worker/all writes to the database to be in a separate process

Reply:
That makes sense.